diffusion/models/stable_diffusion.py

```diff
@@ -91,6 +90,9 @@ def __init__(self,
         self.vae = vae
         self.noise_scheduler = noise_scheduler
         self.loss_fn = loss_fn
+        if prediction_type.lower() not in ['sample', 'epsilon', 'v_prediction']:
+            raise ValueError(f'prediction type must be one of sample, epsilon, or v_prediction. Got {prediction_type}')
+        self.prediction_type = prediction_type.lower()
```

Comment:
```suggestion        
        self.prediction_type = prediction_type.lower()
        if prediction_type not in ['sample', 'epsilon', 'v_prediction']:
            raise ValueError(f'prediction type must be one of sample, epsilon, or v_prediction. Got {prediction_type}')
```
nit a little cleaner to move lower to top and do once

Reply:
Done!